pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentStickyKeyDispatcherMultipleConsumers.java

```diff
@@ -183,22 +180,6 @@ public synchronized void removeConsumer(Consumer consumer) throws BrokerServiceE
         }
     }
 
-    private static final FastThreadLocal<Map<Consumer, List<Entry>>> localGroupedEntries =
```

Comment:
> - Remove the unnecessary thread-local fields localGroupedEntries and the unused localGroupedPositions.
>   - It is unnecessary to eliminate the creation of short-lived objects in modern JVMs. It just clutters the code without any performance benefit. "No garbage" style is relevant mainly for long-lived objects stored in caches or large array allocations.

I am thinking about whether the "No garbage" design of Netty is unnecessary.

Reply:
Since we had this optimization before, we should leave them there. There are many cases using `recycle objects` in our project, if this place can be removed, do you think that others can also be removed?

Reply:
> if this place can be removed, do you think that others can also be removed?

Recycling is a waste for short-lived objects in modern JVMs. In those cases, the JVM can optimize the allocations and memory management so that it doesn't really add essential GC overhead. Recycling has higher overhead in those cases. For long living objects, there's a potential benefit. There are other benchmarks from a few years ago in the context of Netty and the performance benefit of recycling over not recycling was very marginal (<10%). Generational ZGC is even better so it might be an unnecessary overhead to do the recycling in most cases. We'd have to confirm with benchmarking in some cases.

Reply:
I'm considering this part resolved.